syntax = "proto3";

package trainer;

service TrainerServer {
    rpc start_training(TrainerServerRequest) returns (TrainerServerResponse) {}
}

message TrainerServerRequest {
    string model_id = 1;
    TrainingHyperparameters hyperparameters = 2;
    string checkpoint_path = 3;
    map<string, int32> model_configuration = 4;
    Data data_info = 5;
}

message TrainerServerResponse {
    int32 training_id = 1;
}

message TrainingHyperparameters {
    int32 batch_size = 1;
    float learning_rate = 2;
}

message Data {
    string dataset_id = 1;
    int32 num_dataloaders = 2;
}